### PR TITLE
Adding codeowners file creation script

### PR DESCRIPTION
**Pull request by SonOfLope**
This pull request gives the following : 
- update ruleset for branch protection to require review from code owners
- sh script that creates a CODEOWNERS file in repo's which names end with "backend", "frontend" or "db". Also adds devops teams as code owners for all .github folder and Dockerfiles/docker-compose files across all repositories.

### CODE REVIEW

**Review by rngadam — CHANGES_REQUESTED** [review]
Seems like the repository devops itself is lacking the Github Workflows to lint markdown and would need linter setup for shell scripts.

**Comment by SonOfLope** [comment]
Shellcheck will be used with the action [shell-linter](https://github.com/marketplace/actions/shell-linter). 
I applied Shellcheck for the current pull request to make sure it passes the check. The following [issue](https://github.com/ai-cfia/github-workflows/issues/99) will add the reusable action workflow.

File: github-management-script/codeowners-file-creation.sh
Comment (rngadam) [inline github-management-script/codeowners-file-creation.sh]: there's a missing EOF newline here but no error from the Github check. However, repo standards validation is setup and there's a validation of EOF newlines here:

https://github.com/ai-cfia/github-workflows/blob/7fd0e183968b9ea1590d4c4614bd91d61331513f/.github/workflows/workflow-repo-standards-validation.yml#L174

not sure what is going on. I"ve seen this also in another pull request where the checks don't seem to be actually running.

**Review by rngadam — CHANGES_REQUESTED** [review]

**Comment by SonOfLope** [comment]
Added sh check workflow. The pipeline wont work for now since we need to wait until the workflow gets merged on [github-workflow. ](https://github.com/ai-cfia/github-workflows/pull/100)

As for the newline, the workflow seems to run but doesnt detect that the file doesnt have new line : 
image
I will investigate.

**Comment by rngadam** [comment]
> I will investigate.

Suggest referencing the issue into which the investigation / fix is being done. Also good place to use "Reference in new issue".

**Review by rngadam — APPROVED** [review]
see comment on linking to related issues.

**Pull request merged**
